test/integration/stats/services_test.rb

```diff
@@ -53,64 +51,59 @@ def teardown
 
 
   end
 
+  test 'with simple plan and cinstance, retrieve data with account timezone' do
+    Timecop.freeze(Time.utc(2009, 12, 11, 19, 10))
+    plan = FactoryBot.create(:application_plan, issuer: @service)
+    @cinstance = FactoryBot.create(:cinstance, plan: plan)
+    @provider_account.update(timezone: 'Madrid')
+
+    provider_login_with @provider_account.admins.first.username, 'supersecret'
+    opts = { period: 'day', metric_name: @metric.system_name }
+    get usage_stats_api_services_path(@cinstance.service_id, format: :json), params: opts
+
+    assert_response :success
+    assert_content_type 'application/json'
+
+    assert_json_contains "period" => { "name" => "day",
+                                       "granularity" => "hour",
+                                       "since" => Time.zone.parse("2009-12-10T20:00:00+01:00"),
+                                       "until" => Time.zone.parse("2009-12-11T20:59:59+01:00"),
+                                       "timezone" => "Europe/Madrid" }
+  end
+
+  test 'with simple plan and cinstance, retrieve data with specified timezone' do
+    Timecop.freeze(Time.utc(2009, 12, 11, 19, 10))
+    plan = FactoryBot.create(:application_plan, issuer: @service)
+    @cinstance = FactoryBot.create(:cinstance, plan: plan)
 
-  context 'with simple plan and cinstance' do
-    setup do
-      plan = FactoryBot.create(:application_plan, issuer: @service)
-      @cinstance = FactoryBot.create(:cinstance, plan: plan)
-      provider_login_with @provider_account.admins.first.username, 'supersecret'
-      Timecop.freeze(Time.utc(2009, 12, 11, 19, 10))
-    end
-
-    should 'retrieve data with account timezone' do
-      @provider_account.update(timezone: 'Madrid')
-
-      opts = { period: 'day', metric_name: @metric.system_name }
-      get usage_stats_api_services_path(@cinstance.service_id, format: :json), params: opts
-
-      assert_response :success
-      assert_content_type 'application/json'
-
-      assert_json_contains(
-       "period" => {"name"=>"day",
-       "granularity"=>"hour",
-       "since"=>Time.zone.parse("2009-12-10T20:00:00+01:00"),
-       "until"=>Time.zone.parse("2009-12-11T20:59:59+01:00"),
-       "timezone"=>"Europe/Madrid"}
-      )
-    end
-
-    should 'retrieve data with specified timezone' do
-      opts = { period: 'day', metric_name: @metric.system_name, timezone: 'Kamchatka' }
-      get usage_stats_api_services_path(@cinstance.service_id, format: :json), params: opts
-
-      assert_response :success
-      assert_content_type 'application/json'
-      assert_json_contains(
-      "period" => {"name"=>"day",
-      "granularity"=>"hour",
-      "since"=>Time.zone.parse("2009-12-11T07:00:00+12:00"),
-      "until"=>Time.zone.parse("2009-12-12T07:59:59+12:00"),
-      "timezone"=>"Asia/Kamchatka"}
-      )
-    end
+    provider_login_with @provider_account.admins.first.username, 'supersecret'
```

Comment:
These seem identical with `with simple plan and cinstance, retrieve data with account timezone`, perhaps sub-class them?

Reply:
You're right. I've spotted another line shared by all test cases that I'll move to the setup.

Reply:
And the problem with grouping in sub-classes is that either they inherit from the main class, sharing setup/teardown and *running every test twice* or they inherit from `ActionDispatch::IntegrationTest` and we need duplicate setups

Reply:
Exactly this

> And the problem with grouping in sub-classes is that either they inherit from the main class, sharing setup/teardown and running every test twice or they inherit from ActionDispatch::IntegrationTest and we need duplicate setups

Subclassing is sometimes not what you want to do, let's not be too picky and accept even if you have to duplicate things.
Remember the goal is to get rid of shoulda, not refactoring nicely. Of course if can make it better, this is always welcome.